Airport Ticket Booking/Models/ApplicationServices.cs

```diff
@@ -0,0 +1,355 @@
+using System.Globalization;
+using Airport_Ticket_Booking.Models.flight;
+using Airport_Ticket_Booking.Models.user;
+
+namespace Airport_Ticket_Booking.Models
+{
+    public delegate void PrintMenuDelegate();
+
+    public class ApplicationServices
+    {
+        private readonly IFlightServices _flightServices;
+        private readonly IUserServices _userServices;
+        private User? _user;
+
+        private static readonly Lock Lock = new();
+        private static ApplicationServices? _instance;
+
+        private ApplicationServices(IFlightServices flightServices, IUserServices userServices)
+        {
+            _flightServices = flightServices;
+            _userServices = userServices;
+        }
+
+        public static ApplicationServices GetInstance(IFlightServices flightServices, IUserServices userServices)
+        {
+            lock (Lock)
+            {
+                _instance ??= new ApplicationServices(flightServices, userServices);
+            }
+
+            return _instance;
+        }
+
+        public bool SignIn(string username, string password)
+        {
+            User? user = _userServices.SignIn(username, password);
+            if (user != null)
+            {
+                _user = user;
+                return true;
+            }
+
+            Console.WriteLine("Invalid username or password.");
+            return false;
+        }
+
+        public bool SignUp(string username, string password)
+        {
+            return _userServices.SignUp(username, password);
+        }
+
+        private int Book(int flightId)
+        {
+            if (_user == null)
+            {
+                return 1;
+            }
+
+            return _flightServices.BookFlight(flightId, _user.UserId) ? 0 : 2;
```

Comment:
Please use enum instead of int to enhance the code readability
Please do that in all places